src/src/lms/lms.service.ts

```diff
@@ -0,0 +1,435 @@
+import { Injectable } from '@nestjs/common';
+import { HasuraService } from 'src/services/hasura/hasura.service';
+import { UserService } from 'src/user/user.service';
+import { HasuraService as HasuraServiceFromServices } from '../services/hasura/hasura.service';
+import { LSMTestTrackingDto } from './dto/lms-test-tracking.dto';
+import { ConfigService } from '@nestjs/config';
+import { SearchLMSDto } from './dto/search-lms.dto';
+
+@Injectable()
+export class LMSService {
+	public table = 'events';
+	public fillable = [
+		'context',
+		'context_id',
+		'created_by',
+		'end_date',
+		'end_time',
+		'location',
+		'location_type',
+		'start_date',
+		'start_time',
+		'updated_by',
+		'user_id',
+	];
+	public returnFields = [
+		'id',
+		'name',
+		'context',
+		'context_id',
+		'created_by',
+		'end_date',
+		'end_time',
+		'type',
+		'location',
+		'master_trainer',
+		'location_type',
+		'start_date',
+		'start_time',
+		'updated_by',
+		'user_id',
+		'reminders',
+	];
+
+	public attendanceReturnFields = [
+		'id',
+		'user_id',
+		'context_id',
+		'created_by',
+		'context',
+		'status',
+		'lat',
+		'long',
+		'rsvp',
+		'photo_1',
+		'photo_2',
+		'date_time',
+		'updated_by',
+	];
+	constructor(
+		private readonly hasuraService: HasuraService,
+		private hasuraServiceFromServices: HasuraServiceFromServices,
+		private readonly userService: UserService,
+	) {}
+
+	public async getTestAllowStatus(header, response) {
+		const userDetail: any = await this.userService.ipUserInfo(header);
```

Comment:
you can get userid from req.mw_user_id , chk MW once @rushi-tekdi and update code accordingly

Reply:
later remove passing `header` from all methods if not needed

Reply:
changed